### PR TITLE
virttest.utils_conn: Add qemu native TLS support

### PR DESCRIPTION
**Pull request by dzhengfy**
Qemu provides a native TLS support now. This change is to add this
support based on existing implementation for libvirt supporting TLS. The
main changes are the different file naming style comparing to libvirt,
such as 'ca-cert.pem' instead of 'cacert.pem'. Another change is some
unnecessary server configrations are removed.

Signed-off-by: Dan Zheng <dzheng@redhat.com>

### CODE REVIEW

File: virttest/utils_conn.py [hunk 1]
```diff
@@ -1025,7 +1043,8 @@ def server_setup(self):
             except remote.SCPError, detail:
                 raise ConnSCPError('AdminHost', local_path,
                                    server_ip, remote_path, detail)
-
+        if self.qemu_tls:
```
Comment (chunfuwen) [inline virttest/utils_conn.py]: add one comment why it need return

File: virttest/utils_conn.py [hunk 2]
```diff
@@ -795,13 +795,16 @@ class TLSConnection(ConnectionBase):
...
+    qemu_tls: True for qemu native TLS support
     """
     __slots__ = ('server_cn', 'client_cn', 'ca_cn', 'CERTTOOL', 'pki_CA_dir',
                  'libvirt_pki_dir', 'libvirt_pki_private_dir', 'client_hosts',
                  'server_libvirtdconf', 'server_syslibvirtd', 'auth_tls',
                  'tls_port', 'listen_addr', 'tls_allowed_dn_list',
                  'custom_pki_path', 'tls_verify_cert', 'tls_sanity_cert',
-                 'ca_cakey_path', 'scp_new_cacert', 'restart_libvirtd')
+                 'ca_cakey_path', 'scp_new_cacert', 'restart_libvirtd',
+                 'name_dict', 'qemu_tls')
```
Comment (chunfuwen) [inline virttest/utils_conn.py]: maybe rename name_dict to more meaningful name

Comment (dzhengfy) [reply virttest/utils_conn.py]: @chunfuwen , update the variable name to 'credential_name'

File: virttest/utils_conn.py (hunk 1 above)
Comment (dzhengfy) [reply virttest/utils_conn.py]: @chunfuwen , added the comment.

**Comment by dzhengfy** [comment]
# avocado run --vt-type libvirt virsh.migrate_options_shared..native_tls.with_postcopy
JOB ID     : 006c9e679741b56ef0e6dd3af018da72ea942c34
JOB LOG    : /root/avocado/job-results/job-2017-09-29T17.57-006c9e6/job.log
 (1/1) type_specific.io-github-autotest-libvirt.virsh.migrate_options_shared.positive_test.native_tls.with_postcopy: PASS (246.24 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 252.86 s

**Comment by dzhengfy** [comment]
# avocado run --vt-type libvirt virsh.migrate_vm..with_tls
JOB ID     : 313bd0c8a2198a1c8dfc89f7095f671c4668e84c
JOB LOG    : /root/avocado/job-results/job-2017-09-29T18.05-313bd0c/job.log
 (01/11) type_specific.io-github-autotest-libvirt.virsh.migrate_vm.positive_testing.p2p_migration.basic.with_tls.with_postcopy: PASS (71.08 s)
 (02/11) type_specific.io-github-autotest-libvirt.virsh.migrate_vm.positive_testing.p2p_migration.basic.with_tls.without_postcopy: PASS (74.87 s)
 (03/11) type_specific.io-github-autotest-libvirt.virsh.migrate_vm.positive_testing.p2p_migration.listen_address.with_tls.with_postcopy: PASS (75.47 s)
 (04/11) type_specific.io-github-autotest-libvirt.virsh.migrate_vm.positive_testing.p2p_migration.listen_address.with_tls.without_postcopy: PASS (74.80 s)
 (05/11) type_specific.io-github-autotest-libvirt.virsh.migrate_vm.positive_testing.tunnelled_migration.basic.with_tls.without_postcopy: PASS (74.72 s)
 (06/11) type_specific.io-github-autotest-libvirt.virsh.migrate_vm.positive_testing.migration_with_ipv6.with_tls.with_postcopy: SKIP
 (07/11) type_specific.io-github-autotest-libvirt.virsh.migrate_vm.positive_testing.migration_with_ipv6.with_tls.without_postcopy: SKIP
 (08/11) type_specific.io-github-autotest-libvirt.virsh.migrate_vm.negative_testing.p2p_migration.unreachable_destenation.with_tls.with_postcopy: PASS (41.29 s)
 (09/11) type_specific.io-github-autotest-libvirt.virsh.migrate_vm.negative_testing.p2p_migration.unreachable_destenation.with_tls.without_postcopy: PASS (45.81 s)
 (10/11) type_specific.io-github-autotest-libvirt.virsh.migrate_vm.negative_testing.p2p_migration.invalid_listen_address.with_tls.with_postcopy: PASS (46.21 s)
 (11/11) type_specific.io-github-autotest-libvirt.virsh.migrate_vm.negative_testing.p2p_migration.invalid_listen_address.with_tls.without_postcopy: PASS (45.56 s)

Skipped cases are due to configuration missing.

**Pull request merged**